src/Install/Requirements.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @file
+ * Contains Class Requirements.
+ *
+ * This file is about checks for requirements during install of MongoDB module.
+ */
+
+namespace Drupal\mongodb\Install;
+
+/**
+ * Class Requirements implements hook_requirements().
+ */
+class Requirements {
+  public function __construct() {
+    $this->extension_name = '';
+    $this->requirements = array();
+
+    // Ensure translations don't break at install time.
+    $this->t = get_t();
+  }
+
+/**
+ * Common entry point for all requirement checks.
+ */
+  public static function check($phase) {
+    // Check extension version
+    $this->checkExtension();
+
+    // Check server version
+    $this->checkServer($phase);
+
```

Comment:
These two "check*" methods both perform the check and modify the instance state. Better let them perform the check and return a result, then have this result be assign in this method. It's a case of the single responsibility principle: the check* do the checks without mutating the instance, and this method only mutates the instance on its own, without performing anything else itself.

Reply:
Have done rest of the comments except this. Though this is the correct principle to be followed currently in checkExtension there are 5 use cases which would need 5 different return types and in checkServer we have close to 3 return data to be returned back to this function.

      I hope you get what I mean. If so I have below in mind.

1. Return numeric values for each case to the calling function
2. Use a switch case maybe in this calling function to add the detailed to $requirements array as currently done in these function

However in this case we would be introducing more conditionals in our code. Just came to my mind.